Repl crash due to SI-9089

See: https://issues.scala-lang.org/browse/SI-9089

```
> def f(i: => Int) = (i, i)
java.lang.AssertionError: assertion failed: List(method apply$mcI$sp, method apply$mcI$sp, method apply$mcI$sp, method apply$mcI$sp, method apply$mcI$sp, method apply$mcI$sp)
	scala.reflect.internal.Symbols$Symbol.suchThat(Symbols.scala:1916)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.matchingSymbolInPrefix$1(SpecializeTypes.scala:1460)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.transformSelect$1(SpecializeTypes.scala:1483)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.transform1(SpecializeTypes.scala:1579)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2$$anonfun$transform$3.apply(SpecializeTypes.scala:1428)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2$$anonfun$transform$3.apply(SpecializeTypes.scala:1428)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.reportError(SpecializeTypes.scala:1420)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.transform(SpecializeTypes.scala:1428)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.transform(SpecializeTypes.scala:1371)
	scala.reflect.internal.Trees$class.itransform(Trees.scala:1347)
	scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:16)
	scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:16)
	scala.reflect.api.Trees$Transformer.transform(Trees.scala:2555)
	scala.tools.nsc.transform.TypingTransformers$TypingTransformer.transform(TypingTransformers.scala:44)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.transform1(SpecializeTypes.scala:1712)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2$$anonfun$transform$3.apply(SpecializeTypes.scala:1428)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2$$anonfun$transform$3.apply(SpecializeTypes.scala:1428)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.reportError(SpecializeTypes.scala:1420)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.transform(SpecializeTypes.scala:1428)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.transform(SpecializeTypes.scala:1371)
	scala.reflect.api.Trees$Transformer$$anonfun$transformTrees$1.apply(Trees.scala:2559)
	scala.reflect.api.Trees$Transformer$$anonfun$transformTrees$1.apply(Trees.scala:2559)
	scala.collection.immutable.List.loop$1(List.scala:173)
	scala.collection.immutable.List.mapConserve(List.scala:189)
	scala.reflect.api.Trees$Transformer.transformTrees(Trees.scala:2559)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.transformNew$1(SpecializeTypes.scala:1518)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.transform1(SpecializeTypes.scala:1524)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2$$anonfun$transform$3.apply(SpecializeTypes.scala:1428)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2$$anonfun$transform$3.apply(SpecializeTypes.scala:1428)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.reportError(SpecializeTypes.scala:1420)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.transform(SpecializeTypes.scala:1428)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.transform(SpecializeTypes.scala:1371)
	scala.reflect.internal.Trees$$anonfun$itransform$2.apply(Trees.scala:1363)
	scala.reflect.internal.Trees$$anonfun$itransform$2.apply(Trees.scala:1361)
	scala.reflect.api.Trees$Transformer.atOwner(Trees.scala:2600)
	scala.tools.nsc.transform.TypingTransformers$TypingTransformer.atOwner(TypingTransformers.scala:30)
	scala.tools.nsc.transform.TypingTransformers$TypingTransformer.atOwner(TypingTransformers.scala:25)
	scala.tools.nsc.transform.TypingTransformers$TypingTransformer.atOwner(TypingTransformers.scala:17)
	scala.reflect.internal.Trees$class.itransform(Trees.scala:1360)
	scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:16)
	scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:16)
	scala.reflect.api.Trees$Transformer.transform(Trees.scala:2555)
	scala.tools.nsc.transform.TypingTransformers$TypingTransformer.transform(TypingTransformers.scala:44)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.transform1(SpecializeTypes.scala:1692)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2$$anonfun$transform$3.apply(SpecializeTypes.scala:1428)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2$$anonfun$transform$3.apply(SpecializeTypes.scala:1428)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.reportError(SpecializeTypes.scala:1420)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.transform(SpecializeTypes.scala:1428)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.transform(SpecializeTypes.scala:1371)
	scala.reflect.api.Trees$Transformer$$anonfun$transformTrees$1.apply(Trees.scala:2559)
	scala.reflect.api.Trees$Transformer$$anonfun$transformTrees$1.apply(Trees.scala:2559)
	scala.collection.immutable.List.loop$1(List.scala:173)
	scala.collection.immutable.List.mapConserve(List.scala:189)
	scala.reflect.api.Trees$Transformer.transformTrees(Trees.scala:2559)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2$$anonfun$transformTemplate$1$1.apply(SpecializeTypes.scala:1599)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2$$anonfun$transformTemplate$1$1.apply(SpecializeTypes.scala:1599)
	scala.reflect.api.Trees$Transformer.atOwner(Trees.scala:2600)
	scala.tools.nsc.transform.TypingTransformers$TypingTransformer.atOwner(TypingTransformers.scala:30)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.transformTemplate$1(SpecializeTypes.scala:1599)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.transform1(SpecializeTypes.scala:1601)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2$$anonfun$transform$3.apply(SpecializeTypes.scala:1428)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2$$anonfun$transform$3.apply(SpecializeTypes.scala:1428)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.reportError(SpecializeTypes.scala:1420)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.transform(SpecializeTypes.scala:1428)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.transform(SpecializeTypes.scala:1371)
	scala.reflect.api.Trees$Transformer.transformTemplate(Trees.scala:2563)
	scala.reflect.internal.Trees$$anonfun$itransform$4.apply(Trees.scala:1408)
	scala.reflect.internal.Trees$$anonfun$itransform$4.apply(Trees.scala:1407)
	scala.reflect.api.Trees$Transformer.atOwner(Trees.scala:2600)
	scala.tools.nsc.transform.TypingTransformers$TypingTransformer.atOwner(TypingTransformers.scala:30)
	scala.tools.nsc.transform.TypingTransformers$TypingTransformer.atOwner(TypingTransformers.scala:25)
	scala.tools.nsc.transform.TypingTransformers$TypingTransformer.atOwner(TypingTransformers.scala:17)
	scala.reflect.internal.Trees$class.itransform(Trees.scala:1406)
	scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:16)
	scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:16)
	scala.reflect.api.Trees$Transformer.transform(Trees.scala:2555)
	scala.tools.nsc.transform.TypingTransformers$TypingTransformer.transform(TypingTransformers.scala:44)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.transform1(SpecializeTypes.scala:1712)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2$$anonfun$transform$3.apply(SpecializeTypes.scala:1428)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2$$anonfun$transform$3.apply(SpecializeTypes.scala:1428)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.reportError(SpecializeTypes.scala:1420)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.transform(SpecializeTypes.scala:1428)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.transform(SpecializeTypes.scala:1371)
	scala.reflect.api.Trees$Transformer$$anonfun$transformStats$1.apply(Trees.scala:2589)
	scala.reflect.api.Trees$Transformer$$anonfun$transformStats$1.apply(Trees.scala:2587)
	scala.collection.immutable.List.loop$1(List.scala:173)
	scala.collection.immutable.List.mapConserve(List.scala:189)
	scala.reflect.api.Trees$Transformer.transformStats(Trees.scala:2587)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2$$anonfun$transform1$6.apply(SpecializeTypes.scala:1585)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2$$anonfun$transform1$6.apply(SpecializeTypes.scala:1583)
	scala.reflect.api.Trees$Transformer.atOwner(Trees.scala:2600)
	scala.tools.nsc.transform.TypingTransformers$TypingTransformer.atOwner(TypingTransformers.scala:30)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.transform1(SpecializeTypes.scala:1583)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2$$anonfun$transform$3.apply(SpecializeTypes.scala:1428)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2$$anonfun$transform$3.apply(SpecializeTypes.scala:1428)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.reportError(SpecializeTypes.scala:1420)
	scala.tools.nsc.transform.SpecializeTypes$$anon$2.transform(SpecializeTypes.scala:1428)
	scala.tools.nsc.transform.SpecializeTypes$SpecializationTransformer$$anonfun$54.apply(SpecializeTypes.scala:1949)
	scala.tools.nsc.transform.SpecializeTypes$SpecializationTransformer$$anonfun$54.apply(SpecializeTypes.scala:1949)
	scala.reflect.internal.SymbolTable.enteringPhase(SymbolTable.scala:235)
	scala.reflect.internal.SymbolTable.exitingPhase(SymbolTable.scala:256)
	scala.tools.nsc.transform.SpecializeTypes$SpecializationTransformer.transform(SpecializeTypes.scala:1949)
	scala.tools.nsc.transform.SpecializeTypes$SpecializationTransformer.transform(SpecializeTypes.scala:1945)
	scala.tools.nsc.ast.Trees$Transformer.transformUnit(Trees.scala:147)
	scala.tools.nsc.transform.Transform$Phase.apply(Transform.scala:30)
	scala.tools.nsc.Global$GlobalPhase$$anonfun$applyPhase$1.apply$mcV$sp(Global.scala:441)
	scala.tools.nsc.Global$GlobalPhase.withCurrentUnit(Global.scala:432)
	scala.tools.nsc.Global$GlobalPhase.applyPhase(Global.scala:441)
	scala.tools.nsc.Global$GlobalPhase$$anonfun$run$1.apply(Global.scala:399)
	scala.tools.nsc.Global$GlobalPhase$$anonfun$run$1.apply(Global.scala:399)
	scala.collection.Iterator$class.foreach(Iterator.scala:750)
	scala.collection.AbstractIterator.foreach(Iterator.scala:1202)
	scala.tools.nsc.Global$GlobalPhase.run(Global.scala:399)
	scala.tools.nsc.Global$Run.compileUnitsInternal(Global.scala:1500)
	scala.tools.nsc.Global$Run.compileUnits(Global.scala:1487)
	scala.tools.nsc.Global$Run.compileSources(Global.scala:1482)
	scala.tools.nsc.Global$Run.compileFiles(Global.scala:1569)
	ammonite.repl.interp.Compiler$$anon$4.compile(Compiler.scala:147)
	ammonite.repl.interp.Interpreter$$anonfun$3$$anonfun$apply$10.apply(Interpreter.scala:145)
	ammonite.repl.interp.Interpreter$$anonfun$3$$anonfun$apply$10.apply(Interpreter.scala:145)
	ammonite.repl.interp.Evaluator$$anon$3.evalClass(Evaluator.scala:138)
	ammonite.repl.interp.Evaluator$$anon$3$$anonfun$processLine$1$$anonfun$apply$14.apply(Evaluator.scala:192)
	ammonite.repl.interp.Evaluator$$anon$3$$anonfun$processLine$1$$anonfun$apply$14.apply(Evaluator.scala:191)
	ammonite.repl.Catching.flatMap(Util.scala:73)
	ammonite.repl.interp.Evaluator$$anon$3$$anonfun$processLine$1.apply(Evaluator.scala:191)
	ammonite.repl.interp.Evaluator$$anon$3$$anonfun$processLine$1.apply(Evaluator.scala:190)
	ammonite.repl.Res$Success.flatMap(Util.scala:22)
	ammonite.repl.interp.Evaluator$$anon$3.processLine(Evaluator.scala:190)
	ammonite.repl.interp.Interpreter$$anonfun$processLine$2$$anonfun$apply$3.apply(Interpreter.scala:42)
	ammonite.repl.interp.Interpreter$$anonfun$processLine$2$$anonfun$apply$3.apply(Interpreter.scala:37)
	ammonite.repl.Res$Success.flatMap(Util.scala:22)
	ammonite.repl.interp.Interpreter$$anonfun$processLine$2.apply(Interpreter.scala:37)
	ammonite.repl.interp.Interpreter$$anonfun$processLine$2.apply(Interpreter.scala:33)
	ammonite.repl.Catching.flatMap(Util.scala:73)
	ammonite.repl.interp.Interpreter.processLine(Interpreter.scala:33)
	ammonite.repl.Repl$$anonfun$action$1$$anonfun$apply$4.apply(Repl.scala:45)
	ammonite.repl.Repl$$anonfun$action$1$$anonfun$apply$4.apply(Repl.scala:44)
	ammonite.repl.frontend.Scoped$$anonfun$flatMap$1.apply(Signaller.scala:43)
	ammonite.repl.frontend.Signaller.apply(Signaller.scala:28)
	ammonite.repl.frontend.Scoped$class.flatMap(Signaller.scala:43)
	ammonite.repl.frontend.Signaller.flatMap(Signaller.scala:10)
	ammonite.repl.Repl$$anonfun$action$1.apply(Repl.scala:44)
	ammonite.repl.Repl$$anonfun$action$1.apply(Repl.scala:43)
	ammonite.repl.Res$Success.flatMap(Util.scala:22)
	ammonite.repl.Repl.action(Repl.scala:43)
	ammonite.repl.Repl.loop$1(Repl.scala:54)
	ammonite.repl.Repl.run(Repl.scala:57)
	ammonite.repl.Repl$.run(Repl.scala:83)
	a$$anon$1.<init>(a:8)
	a$.main(a:2)
	a.main(a)
	sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	java.lang.reflect.Method.invoke(Method.java:497)
	sbt.Run.invokeMain(Run.scala:67)
	sbt.Run.run0(Run.scala:61)
	sbt.Run.sbt$Run$$execute$1(Run.scala:51)
	sbt.Run$$anonfun$run$1.apply$mcV$sp(Run.scala:55)
	sbt.Run$$anonfun$run$1.apply(Run.scala:55)
	sbt.Run$$anonfun$run$1.apply(Run.scala:55)
	sbt.Logger$$anon$4.apply(Logger.scala:85)
	sbt.TrapExit$App.run(TrapExit.scala:248)
	java.lang.Thread.run(Thread.java:745)
Something unexpected went wrong =(
```

--- a/build.sbt
+++ b/build.sbt
@@ -13,7 +13,7 @@ publishArtifact := false
 publishTo := Some(Resolver.file("Unused transient repository", file("target/unusedrepo")))
 
 val sharedSettings = Seq(
-  scalaVersion := "2.11.5",
+  scalaVersion := "2.11.6",
   organization := "com.lihaoyi",
   version := "0.3.0-SNAPSHOT",
   libraryDependencies += "com.lihaoyi" %% "utest" % "0.3.0" % "test",

